### ISSUE TITLE
Links for sharing SageMathCell output no longer active

### ISSUE THREAD

**Issue by paulmasson** [sagemath/sagecell#470]
Clicking the "Share" button previously produced active anchors for permalinks and temporary links. Presumably their deactivation has something to do with the most recent commit (https://github.com/sagemath/sagecell/commit/c633e68c5912e7902dab72cb9f037801af45ef37).

**Comment by novoselt** [comment]
Will investigate - seemed to work fine after the switch yesterday, but I got another report about the same time as yours and at the moment I also have the same problem.

**Comment by novoselt** [comment]
The problem is with cells containing unicode (again). Such a request prevents any further modifications to the database, although reads are fine (so old permalinks are working). The error we are getting is
```
ProgrammingError: (sqlite3.ProgrammingError) You must not use 8-bit bytestrings unless you use a text_factory that can interpret 8-bit bytestrings (like text_factory = str). It is highly recommended that you instead just switch your application to Unicode strings.
```
Unfortunately I can't work further on it at the moment - will try again tomorrow evening and either fix it or revert back to the old server for now. The most recent commit is unlikely to be the cause - it affected the way server process starts/quits and it works just fine. But something has changed with sqlite and related packages in 16.04.

If someone wants to dig deeper meanwhile, the relevant file is probably https://github.com/sagemath/sagecell/blob/master/db_sqlalchemy.py

**Comment by novoselt** [comment]
OK, I believe it is fixed now. Over a year ago there was a problem with using Unicode characters and I fixed it via d0833cd1b2cba0fb652beac9d2f8f490f9aaf613. As I have figured out now, the problem was decoding twice - once on a compute server and once on the database one. Back then I have only updated all compute servers (which stopped decoding) while the database server kept happily running an older version doing one necessary decoding. With the recent migration the database server stopped doing it as well, hence the issues. Those who have favourite Unicode permalinks and interacts - please report any new (or old) issues!
